homework9/hw1.py

```diff
@@ -0,0 +1,120 @@
+"""
+Write a function that merges integer from sorted files and returns an iterator
+
+file1.txt:
+1
+3
+5
+
+file2.txt:
+2
+4
+6
+
+# >>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
+[1, 2, 3, 4, 5, 6]
+"""
+from contextlib import ExitStack
+from pathlib import Path
+from typing import Iterator, List, Union
+
+
+def translate_file_to_list(file):
+    return list(map(int, file.read().strip().split("\n")))
```

Comment:
You load the file to memory when you building a list. It is better to avoid this.

Reply:
Rewrite it to `[int(line) for line in file]`

Reply:
Rewrite it to
```python
def translate_file_to_list(file):
    output = []
    for line in file:
        output.append(int(line))

    return output
```